dvc/cache/base.py

```diff
@@ -141,25 +141,30 @@ def _do_link(self, from_info, to_info, link_method):
             "Created '%s': %s -> %s", self.cache_types[0], from_info, to_info,
         )
 
+    def add(self, path_info, tree, hash_info):
```

Comment:
This should really be:

```suggestion
    def add(self, object):
```

but that will come later (e.g. an object might be a plain file, a `.dir` or a run cache).